### PR TITLE
MULE-10936: Allow the SchedulerService to create pools per app rather

### PR DESCRIPTION
**Pull request by elrodro83**
that per container
* review

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/scheduler/SchedulerPoolsConfig.java [hunk 1]
```diff
@@ -19,48 +19,48 @@
    * @return the maximum time (in milliseconds) to wait until all tasks in all the runtime thread pools have completed execution
    *         when stopping the scheduler service.
    */
-  long getGracefulShutdownTimeout();
+  Long getGracefulShutdownTimeout();
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/scheduler/SchedulerPoolsConfig.java]: Why not Long -> OptionalInt?

File: services/scheduler/src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java [hunk 2]
```diff
@@ -163,83 +169,83 @@ private ContainerThreadPoolsConfig() {
   }
 
   @Override
-  public long getGracefulShutdownTimeout() {
+  public Long getGracefulShutdownTimeout() {
     return gracefulShutdownTimeout;
   }
 
-  private void setGracefulShutdownTimeout(long gracefulShutdownTimeout) {
+  private void setGracefulShutdownTimeout(Long gracefulShutdownTimeout) {
```
Comment (dfeist) [inline services/scheduler/src/main/java/org/mule/service/scheduler/internal/config/ContainerThreadPoolsConfig.java]: Why do setters need to use a nullable value?

File: services/scheduler/src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java [hunk 3]
```diff
@@ -107,20 +107,21 @@ public SchedulerThreadPools(String name, SchedulerPoolsConfig threadPoolsConfig)
...
-        new ThreadPoolExecutor(threadPoolsConfig.getCpuLightPoolSize(), threadPoolsConfig.getCpuLightPoolSize(),
-                               0, SECONDS, new LinkedBlockingQueue<>(threadPoolsConfig.getCpuLightQueueSize()),
+        new ThreadPoolExecutor(threadPoolsConfig.getCpuLightPoolSize().getAsInt(),
+                               threadPoolsConfig.getCpuLightPoolSize().getAsInt(),
+                               0, SECONDS, new LinkedBlockingQueue<>(threadPoolsConfig.getCpuLightQueueSize().getAsInt()),
                                new SchedulerThreadFactory(cpuLightGroup), byCallerThreadGroupPolicy);
     ioExecutor =
-        new ThreadPoolExecutor(threadPoolsConfig.getIoCorePoolSize(), threadPoolsConfig.getIoMaxPoolSize(),
-                               threadPoolsConfig.getIoKeepAlive(), MILLISECONDS,
+        new ThreadPoolExecutor(threadPoolsConfig.getIoCorePoolSize().getAsInt(), threadPoolsConfig.getIoMaxPoolSize().getAsInt(),
```
Comment (dfeist) [inline services/scheduler/src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java]: What about default if not set?

Comment (elrodro83) [inline services/scheduler/src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java]: we'll handle that when we make values not required

Comment (dfeist) [inline services/scheduler/src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java]: ok, at least it with throw exception, and not just return null.

**Pull request merged**